scripts/babel-plugin-jest-replace-ts-require-assignment.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+// Replace `import thing = require('thing')` with `const thing = require('thing')` which allows us to keep CJS semantics
+
+module.exports = ({template}) => {
+  const moduleExportsDeclaration = template(`
+    const NAME = require('IMPORT');
+  `);
+  return {
+    name: 'jest-replace-ts-require-assignment',
+    visitor: {
+      TSImportEqualsDeclaration(path) {
+        const {node} = path;
+
+        path.replaceWith(
+          moduleExportsDeclaration({
+            IMPORT: node.moduleReference.expression,
+            NAME: node.id,
```

Comment:
Looks like we'll need to make this name unique: https://circleci.com/gh/facebook/jest/70063?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Reply:
Aww. Probably some babel helper for that

Reply:
`path.scope.generateUidIdentifier(node.id.name)` :)
Would usually also suggest `path.scope.generateUidIdentifierBasedOnNode(node)` which magically extracts the appropriate name from the assignment, but not sure if that supports weird `TSImportEqualsDeclaration`s ^^

Reply:
changing to `import blah from 'blah'` made babel's own transformation kick in later, so this should be solved. CI is happy at least 🙂